botocore/args.py

```diff
@@ -543,6 +550,47 @@ def _compute_connect_timeout(self, config_kwargs):
         if connect_timeout:
             config_kwargs['connect_timeout'] = connect_timeout
 
+    def _compute_request_compression_config(self, config_kwargs):
+        min_size = config_kwargs.get('request_min_compression_size_bytes')
+        disabled = config_kwargs.get('disable_request_compression')
+        if min_size is None:
+            min_size = self._config_store.get_config_variable(
+                'request_min_compression_size_bytes'
+            )
+            config_kwargs['request_min_compression_size_bytes'] = min_size
+        # conversion func is skipped so input validation must be done here
+        # regardless if the value is coming from the config store or the
+        # config object
+        self._validate_min_compression_size(min_size)
+        if disabled is None:
+            disabled = self._config_store.get_config_variable(
+                'disable_request_compression'
+            )
+        else:
+            # if the user provided a value we must check if its a boolean
+            disabled = ensure_boolean(disabled)
+        config_kwargs['disable_request_compression'] = disabled
+
+    def _validate_min_compression_size(self, min_size):
+        if min_size is not None:
```

Comment:
This is a bit of a quirk in the current implementation. This value should never be `None` since the config provider provides a default value. Without it, however, a number of test failures happen in `test_client` that do not explicitly set this variable on the config store. To avoid this, we may consider moving this function to `utils.py`, use it as the `conversion_func` in the config provider and call it in `_compute_request_compression_config` if set on the client config.

Reply:
About 3/4 tests in `tests/unit/test_client.py`. The error is pretty much the same for all of them:

```
E           TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'
```

The issue is that the test client is using an empty `ConfigValueStore` object: https://github.com/boto/botocore/blob/9ef2d48793a564d5a637d67f3af817116bfe1aff/tests/unit/test_client.py#L151
so there's nothing to fall back to. We could add defaults here as the config provider normally does, but it didn't seem wise to me to mess with something that is foundational to hundreds of tests.

The exact tests that fail are any in `TestAutoGeneratedClient` and `TestClientErrors` that don't explicitly set `request_min_compression_size_bytes` in either the config store or on a client config object.

Reply:
That explanation makes sense to me. But why does this only surface now, and not for example in #2955 where I added new config variables?

Reply:
I believe it's because no default is expected from the config provider for `sdk_ua_app_id` whereas for `request_compression_min_size_bytes` there is.